src/bot.ts

```diff
@@ -116,56 +118,67 @@ export const crawlPendingTasks = async (): Promise<void> => {
 
                         // Check if the entry is either object or null
-                        if (typeof entry !== 'object' || entry === null) 
+                        if (typeof entry !== 'object' || entry === null)
                             throw new Error(`Entry at index ${index} is not a valid object.`);
 
                         // Separate required fields from the entry
                         const { text, source_url_id, date, contractor } = entry;
 
                         // Check if text field is of type string
-                        if (typeof text !== 'string') 
+                        if (typeof text !== 'string')
                             throw new Error(`Entry at index ${index} has invalid 'text' property.`);
-                        
+
 
                         // Check if sourceUrlId field is of type bigint
-                        if (typeof source_url_id !== 'bigint') 
+                        if (typeof source_url_id !== 'bigint')
                             throw new Error(`Entry at index ${index} has invalid 'sourceUrlId' property.`);
-                        
+
 
                         // Check if date field is of type Date
-                        if (!(date instanceof Date)) 
+                        if (!(date instanceof Date))
                             throw new Error(`Entry at index ${index} has invalid 'date' property.`);
-                        
+
 
                         // Check if contractor field is of type string
-                        if (typeof contractor !== 'string') 
+                        if (typeof contractor !== 'string')
                             throw new Error(`Entry at index ${index} has invalid 'contractor' property.`);
-                    
+
                         // Save the new validated CrawledData object
                         return { text, source_url_id, date, contractor };
                     });
 
-                    // Perform database operations in a single transaction
-                    await prisma.$transaction([
+                    // Filter entries to include only those that match keywords
+                    const filteredEntries = await filterEntriesByKeywords(crawledDataEntries);
+
+                    // Create promises for each entry insertion
+                    const insertPromises = filteredEntries.map((entry) =>
+                        prisma.crawledData.create({
+                            data: entry,
+                        }).catch((error) => {
+
+                            // Display the error
+                            insertCrawledDataEntrieErrorHadler(entry, error.message);
 
-                        // Save crawled data
-                        prisma.crawledData.createMany({
-                            data: crawledDataEntries,
-                        }),
+                            return null; // Skip this entry
+                        })
+                    );
```

Comment:
Why not createMany?

Reply:
If will be way more efficient if we use createMany instead of single create for each entry.